code/game/objects/items/weapons/tools/mods/_upgrades.dm

```diff
@@ -308,6 +308,11 @@
 		G.fire_sound = 'sound/items/bikehorn.ogg'
 	if(weapon_upgrades[GUN_UPGRADE_RIGGED])
 		G.rigged = TRUE
+	if(weapon_upgrades[GUN_UPGRADE_FOREGRIP])
+		G.braceable = 0
+	if(weapon_upgrades[GUN_UPGRADE_BIPOD])
+		G.braceable = 2
+		G.slowdown_hold = 2
```

Comment:
```suggestion
```

Reply:
Doesn't really make sense. Any extra disadvantage should be put on the mod itself, instead of the unique upgrade type. Example, what if we implement a One Star variant that wasn't meant to have disadvantages?